github_connector/models/github_repository_branch.py

```diff
@@ -0,0 +1,304 @@
+        for root, dirs, files in os.walk(existing_folder):
+            if '/.git' not in root:
+                for fic in files:
+                    if fic != '.gitignore':
+                        res.append(os.path.join(root, fic))
+        return res
+
+    @api.model
+    def analyze_code_one(self, branch):
+        """Overload Me in custom Module that manage Source Code analysis.
+        """
+        self.ensure_one()
+        path = branch.local_path
+        # Compute Files Sizes
+        size = 0
+        for file_path in self._get_analyzable_files(path):
+            try:
+                size += os.path.getsize(file_path)
+            except:
+                _logger.warning(
+                    "Warning : unable to eval the size of '%s'." % (file_path))
+
+        try:
+            Repo(path)
+        except:
+            # If it's not a correct repository, we flag the branch
+            # to be downloaded again
+            self.state = 'to_download'
+            return {'size': 0}
+
+        return {'size': size}
+
+    @api.multi
+    def _analyze_code(self):
+        partial_commit = safe_eval(
+            self.env['ir.config_parameter'].get_param(
+                'git.partial_commit_during_analysis'))
+        for branch in self:
+            path = branch.local_path
+            if not os.path.exists(path):
+                _logger.warning(
+                    "Warning Folder %s not found. Analyze skipped." % (path))
+            else:
+                _logger.info("Analyzing Source Code in %s ..." % (path))
+                vals = branch.analyze_code_one(branch)
+                vals.update({
+                    'last_analyze_date': datetime.today(),
+                    'state': 'analyzed',
+                })
+                # Mark the branch as analyzed
+                branch.write(vals)
+                if partial_commit:
+                    self._cr.commit()  # pylint: disable=invalid-commit
+        return True
+
+    # Compute Section
+    @api.multi
+    @api.depends('name', 'repository_id.name')
+    def _compute_complete_name(self):
+        for branch in self:
+            branch.complete_name =\
+                branch.repository_id.complete_name + '/' + branch.name
+
+    @api.multi
+    @api.depends('size')
+    def _compute_mb_size(self):
+        for branch in self:
+            branch.mb_size = float(branch.size) / (1024 ** 2)
+
+    @api.multi
+    @api.depends('organization_id', 'name')
+    def _compute_organization_milestone_id(self):
+        for branch in self:
+            for milestone in branch.organization_id.organization_milestone_ids:
+                if milestone.name == branch.name:
+                    branch.organization_milestone_id = milestone
+
+    @api.multi
+    @api.depends('complete_name')
+    def _compute_local_path(self):
+        source_path = tools.config.get('source_code_local_path', False)
+        for branch in self:
+            branch.local_path = os.path.join(
```

Comment:
Here the execution fails (with a traceback) when _source_path_ is false.
You should add some robustness.